docs/docs/explainers/explainer-writing-noir.md

```diff
@@ -149,6 +161,17 @@ Use `  if is_unconstrained() { /`, to conditionally execute code if being called
 
 Unless you're well into the depth of gate optimization, this advanced section can be ignored.
 
+### `#[no_predicates]` over for "pure" functions
```

Comment:
I'm not sure about advertising `no_predicates` yet as for unexperienced developers it is a bit more dangerous and less obvious about what is happening. @jfecher what do you think?

Reply:
Also we have a few errors such as https://github.com/noir-lang/noir/issues/5672 that still exist so I would rather not advertise publicly yet @jzaki

Reply:
I put this in the advanced section with plenty of caveats.
Whilst the start is more general to help devs new to Noir start to appreciate what optimisations do/don't look like, the Advanced section is caveated ( the line above). I'll make this a warning.

The contents below came up as a question in slack (by a more advanced ecosystem dev), they happened to refer to the form of predicates in passing.